test/integration/puppetlabs/services/jruby/tasks_test.clj

```diff
@@ -0,0 +1,116 @@
+            [puppetlabs.services.jruby.puppet-environments :as puppet-env]
+            [puppetlabs.services.master.master-core :as mc]
+            [me.raynes.fs :as fs]
+            [cheshire.core :as cheshire]
+            [schema.core :as schema]
+            [schema.test :as schema-test]
+            [puppetlabs.puppetserver.testutils :as testutils]
+            [puppetlabs.trapperkeeper.app :as tk-app]
+            [puppetlabs.trapperkeeper.testutils.bootstrap :as tk-bootstrap]))
+
+(use-fixtures :once schema-test/validate-schemas)
+
+(def TaskOptions
+  {:name schema/Str
+   :module-name schema/Str
+   :metadata? schema/Bool
+   :number-of-files schema/Num})
+
+(defn make-task-dir
+  [env-dir mod-name]
+  (fs/file env-dir "modules" mod-name "tasks"))
+
+(schema/defn gen-task
+  "Assumes tasks dir already exists -- generates a set of task files for a
+  single task."
+  [env-dir task :- TaskOptions]
+  (let [task-name (:name task)
+        extensions (cycle '(".rb" "" ".sh" ".exe" ".py"))
+        task-dir (make-task-dir env-dir (:module-name task))]
+    (fs/mkdirs task-dir)
+    (when (:metadata? task)
+      (fs/create (fs/file task-dir (str task-name ".json"))))
+    (dotimes [n (:number-of-files task)]
+      (fs/create (fs/file task-dir (str task-name (nth extensions n ".rb")))))))
+
+(defn gen-tasks
+  "Tasks is a list of task maps, with keys:
+  :name String, file name of task
+  :module-name String, name of module task is in
+  :metadata? Boolean, whether to generate a metadata file
+  :number-of-files Number, how many executable files to generate for the task (0 or more)"
+  [env-dir tasks]
+  (dorun (map (partial gen-task env-dir) tasks)))
+
+(defn create-env
+  [env-dir tasks]
+  (testutils/create-env-conf env-dir "")
+  (gen-tasks env-dir tasks))
+
+(defn expected-tasks-info
+  [tasks]
+  (->> tasks
+       (map (fn [{:keys [name module-name]}]
+              {:module {:name module-name}
+               :name (if (= "init" name)
+                       module-name
+                       (str module-name "::" name))}))))
+
+(deftest ^:integration all-tasks-test
+  (testing "requesting all tasks"
+    (let [code-dir (ks/temp-dir)
+          conf-dir (ks/temp-dir)
+          config (jruby-testutils/jruby-puppet-tk-config
+                   (jruby-testutils/jruby-puppet-config
+                     {:master-code-dir (.getAbsolutePath code-dir)
+                      :master-conf-dir (.getAbsolutePath conf-dir)}))]
+
+      (tk-bootstrap/with-app-with-config
+        app
+        jruby-testutils/jruby-service-and-dependencies
+        config
+        (let [jruby-service (tk-app/get-service app :JRubyPuppetService)
+              instance (jruby-testutils/borrow-instance jruby-service :test)
+              jruby-puppet (:jruby-puppet instance)
+              env-registry (:environment-registry instance)
+              _ (testutils/create-file (fs/file conf-dir "puppet.conf")
```

Comment:
No, that looks fine to me, I think this is an artifact of me breaking up this into two let blocks from the copied test